packages/aws-cdk/lib/cdk-toolkit.ts

```diff
@@ -325,7 +323,21 @@ export class CdkToolkit {
         print("Detected change to '%s' (type: %s) while 'cdk deploy' is still running. " +
             'Will queue for another deployment after this one finishes', filePath, event);
       }
-    });
+    };
+
+    chokidar.watch(watchIncludes, {
+      ignored: watchExcludes,
+      cwd: rootDir,
+      // ignoreInitial: true,
+    }).on('ready', () => {
+      latch = 'open';
+      // trigger a dummy event here to make an intial deployment,
+      // the values sent into workflow are dummy values.
+      // intentionally not awaiting the result so that we can begin to watch and queue changes.
+      void workflow('add', 'initial');
```

Comment:
Hmmm. Forgetting this promise like this doesn't seem great to me. Wouldn't this trigger an "uncaught promise warning" if the initial deployment fails?

Can you try returning this promise from function, and see if that helps?

Reply:
I changed it to what I think you're asking for. My thoughts here are that it feels wrong to make the 'ready' callback async because during that time we would lose track of file changes -- is my instinct there correct?

Reply:
I also tested my own CDK stack (that fails deployment) with `void handleFileChangeEvent` and I received the same "failed to deploy" message that I was expecting:

<img width="810" alt="Screen Shot 2021-12-16 at 1 42 17 PM" src="https://user-images.githubusercontent.com/36202692/146429912-0cbd2443-bd33-4e85-9795-57df159791f6.png">

Reply:
> My thoughts here are that it feels wrong to make the 'ready' callback async because during that time we would lose track of file changes -- is my instinct there correct?

Can you check what the behavior is manually, and report back here? 🙂

Reply:
So here's what I found:

You can specify async callbacks, and typescript will not yell at you, but they will be ignored. see this [issue](https://github.com/paulmillr/chokidar/issues/886). For example, even though the `handleFileChangeEvent` function is async, no one is actually waiting for the event to finish. Same goes for the callback at the `'ready'` event.

As to any unhandled promise warning, I think we are in the clear. It seems related to how we are catching errors in `invokeDeployFromWatch` anyway -- we ignore them, and `deploy` will show them (as I have confirmed locally when using `void handleFileChangeEvent()`).

https://github.com/aws/aws-cdk/blob/264d93175778b71e8bbeb93b40b67d4746d9f899/packages/aws-cdk/lib/cdk-toolkit.ts#L609-L611

All that to say, I think we are fine with `void handleFileChangeEvent()` we are also fine with `return handleFileChangeEvent()`. In both cases, a failed deployment will show on screen as expected, with no unhandled promises.

Reply:
With that in mind I've changed the function to be `async` and we `await handleFileChangeEvent()`. But it is an inconsequential change. Can confirm that the behavior for this is as expected on my local computer -- I get the deployment failure or success message printed out. And `watch` is immediately listening for file changes (not waiting for first deployment to finish).